Seohee/Lv2/2개 이하로 다른 비트.js

```diff
@@ -0,0 +1,13 @@
+function solution(numbers) {
+  const answer = [];
+  function f(x) {
+    //짝수는 다음 수가 1개 차이
+    if (x % 2 === 0) return x + 1;
+    //홀수 체크
+    const bit = "0" + x.toString(2);
+    const index = bit.lastIndexOf("0");
+    return parseInt(`${bit.slice(0, index)}10${bit.slice(index + 2)}`, 2);
+  }
+  for (const number of numbers) answer.push(f(number));
+  return answer;
```

Comment:
```suggestion
  function f(x) {
    //짝수는 다음 수가 1개 차이
    if (x % 2 === 0) return x + 1;
    //홀수 체크
    const bit = "0" + x.toString(2);
    const index = bit.lastIndexOf("0");
    return parseInt(`${bit.slice(0, index)}10${bit.slice(index + 2)}`, 2);
  }
  return numbers.map(number => f(number);
```
map을 활용하면 answer을 선언하지 않고 return을 해줄 수 있고, 이렇게 짜면 map의 콜백함수를 먼저 선언해준 꼴이 돼서
```js
function solution(numbers) {
  return numbers.map((x) => {
    if (x % 2 === 0) return x + 1;

    const bit = "0" + x.toString(2);
    const index = bit.lastIndexOf("0");
      
    return parseInt(`${bit.slice(0, index)}10${bit.slice(index + 2)}`, 2);
  });
}
```
이런식으로도 짤 수 있을거 같아요.